### DISCUSSION TITLE
"Inacessible" badge with self-hosted shields docker image and self-hosted gitlab

### DISCUSSION THREAD

**Discussion by eslowney** [badges/shields#8890] [General]
On our internal LAN, I have a private (non-routable) Gitlab instance with an SSL cert signed by our internal CA.  Projects are "public" in Gitlab, in that they can be viewed via Firefox incognito mode w/o needing to authenticate.  E.g. `https://gitlab.local/project/`.  I mention them being public because they don't need a token to access them (as tested by incognito mode).

I recently stood up a shields.io docker instance by following the instructions, and I port-forwarded 8080, so it's running on the docker server accessible at: `http://docker.local:8080`.

Every combination I've tried of `/endpoint?url=https://gitlab.local/{project}/raw/{branch}/badge.json `or `dynamic/json?url=https://gitlab.local/{project}/raw/{branch}/badge.json` always ends up with the grey `inaccessible` badge.  I can't tell if I'm doing something wrong, or this is an unsupported config.  Shields does URL-encode the slashes and such properly.

I've basically been following the guides/URLs in the below two links, to no avail.  Is there a docker command to enable some sort of debugging output?  Currently the only output I get now is something about Github tokens being exhausted.

https://github.com/badges/shields/issues/2911#issuecomment-604772061

https://medium.com/@iffi33/adding-custom-badges-to-gitlab-a9af8e3f3569#4773

**Comment by calebcartwright** [comment]
There's a few things that could be happening here, and it's obviously a bit of a black box for us.

First, try updating the environment variables within the container where the badge server is running to include `TRACE_SERVICES=true`. This will give you a much more verbose/debugging level output that should shine a light on what's happening with the communication.

**Reply by eslowney** [reply]
Thanks for the debugging tip, I now see that it is also listed in the below ticket #8768 😄 .  Will try that and report back, probably in a week or so.

**Comment by chris48s** [comment]
I wonder if this is related to https://github.com/badges/shields/issues/8768 ?

**Reply by eslowney** [reply]
Thanks for the ticket #.  Seems similar.  I'm using gitlab itself for the native pipeline and coverage badges, but I wanted more badges, hence my discovery of shields.  I don't think I need to use the API, just (preferably) grab key=values from a gitlab-hosted json file.
